.circleci/config.yml

```diff
@@ -51,6 +51,57 @@ jobs:
       - run: wget -O ./bin/manifest-tool https://github.com/estesp/manifest-tool/releases/download/v1.0.0/manifest-tool-linux-amd64 && chmod +x ./bin/manifest-tool
       - run: echo "export GO111MODULE=on" >>$BASH_ENV
       - run: PATH=./bin:$PATH inv push-multiarch --binaries=speaker --tag=${CIRCLE_BRANCH:-${CIRCLE_TAG}} --docker-user=metallb
+  deploy-controller-pr:
+    working_directory: /go/src/go.universe.tf/metallb
+    docker:
+      - image: circleci/golang:1.13
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run: sudo apt-get install python-pip
+      - run: sudo pip install invoke semver pyyaml
+      - run: docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
+      - run: mkdir -p ./bin
+      - run: wget -O ./bin/manifest-tool https://github.com/estesp/manifest-tool/releases/download/v1.0.0/manifest-tool-linux-amd64 && chmod +x ./bin/manifest-tool
+      - run: echo "export GO111MODULE=on" >>$BASH_ENV
+      - run: echo ${CIRCLE_PULL_REQUEST}
+      - run: echo ${CIRCLE_BRANCH}
+      - run: echo ${CIRCLE_TAG}
+      # CIRCLE_PULL_REQUEST is a URL in the form: https://github.com/metallb/metallb/pull/1
+      # We use sed to get the pull request ID out of the URL. CIRCLE_PR_NUMBER would be
+      # more convenient, but unfortunately it is not always available.
+      # https://circleci.com/docs/2.0/env-vars/
+      #
+      # If the job is being run against a PR, use "pr-<ID>" as the image tag and push the
+      # result to metallbdev/controller. Otherwise this job is not expected to run so we
+      # just fail.
+      - run: if [ -n "$CIRCLE_PULL_REQUEST" ]; then PATH=./bin:$PATH inv push-multiarch --binaries=controller --tag=pr-$(echo $CIRCLE_PULL_REQUEST | sed -e 's/.*pull\/\(.*\)/\1/') --docker-user=metallbdev ; else exit 1 ; fi
```

Comment:
using something like: `$(basename "$CIRCLE_PULL_REQUEST")` wouldn't be cleaner to read rather than `sed`?

Reply:
good suggestion, thanks!